tools/exporter/exporter.py

```diff
@@ -610,75 +737,104 @@ def sync_exports(logger, sc_client, settings):
-                        save_exported_document(logger, export_path, export_doc, export_filename, export_format)
-                    elif export_format == 'json':
-                        export_doc = json.dumps(audit_json, indent=4)
-                        save_exported_document(logger, export_path, export_doc, export_filename, export_format)
-                    elif export_format == 'csv':
-                        csv_exporter = csvExporter.CsvExporter(audit_json, export_inactive_items_to_csv)
-                        csv_export_filename = audit_json['template_id']
-                        csv_exporter.append_converted_audit_to_bulk_export_file(os.path.join(export_path, csv_export_filename + '.csv'))
-                    elif export_format == 'media':
-                        media_export_path = os.path.join(export_path, 'media', export_filename)
-                        extension = 'jpg'
-                        media_id_list = get_media_from_audit(logger, audit_json)
-                        for media_id in media_id_list:
-                            logger.info("Saving media_{0} to disc.".format(media_id))
-                            media_file = sc_client.get_media(audit_id, media_id)
-                            media_export_filename = media_id
-                            save_exported_media_to_file(logger, media_export_path, media_file, media_export_filename, extension)
-                    elif export_format == 'web-report-link':
-                        web_report_link = sc_client.get_web_report(audit_id)
-                        web_report_data = [template_id,
-                               csvExporter.get_json_property(audit_json, 'template_data', 'metadata', 'name'),
-                               audit_id,
-                               csvExporter.get_json_property(audit_json, 'audit_data', 'name'),
-                               web_report_link]
-                        save_web_report_link_to_file(logger, export_path, web_report_data)
-                logger.debug('setting last modified to ' + audit['modified_at'])
-                update_sync_marker_file(audit['modified_at'])
-            else:
-                logger.info('Audit\'s modified_at value is less than {0} seconds in the past, skipping for now!'.format(media_sync_offset))
+            process_audit(logger, settings, sc_client, audit)
+            export_count += 1
+
+def check_if_media_sync_offset_satisfied(logger, settings, audit):
+    modified_at = dateutil.parser.parse(audit['modified_at'])
+    now = datetime.utcnow()
+    elapsed_time_difference = (pytz.utc.localize(now) - modified_at)
+    # if the media_sync_offset has been satisfied
+    if not elapsed_time_difference > timedelta(seconds=settings[MEDIA_SYNC_OFFSET_IN_SECONDS]):
+        logger.info('Audit\'s modified_at value is less than {0} seconds in the past, skipping for now!'.format(
+            settings[MEDIA_SYNC_OFFSET_IN_SECONDS]))
```

Comment:
Is it going to be picked up later?

Reply:
Yes, if in looping mode it will keep checking until the offset is satisfied. If not in looping mode it will just exit after logging that message.

Reply:
Ok. Maybe adjust the comment text to something like 'Audit <audit_id> was modified too recently, some media may not have completed syncing. Skipping export until next sync cycle.'